src/server.py

```diff
@@ -23,23 +24,44 @@ def server():
     while True:
         try:
             buffer_length = 8
-            response = u''
-            while response[-2:] != u"\r\n":
+            client_request = u''
+            while client_request[-2:] != u"\r\n":
                 part = conn.recv(buffer_length)
-                response += part.decode('utf8')
-                print('Recieved: ', part)
+                client_request += part.decode('utf8')
 
-            print("Message sent...")
-            conn.sendall(response.encode('utf8'))
+            print(client_request)
+            conn.sendall(response_ok().encode('utf8'))
+            conn.sendall(client_request.encode('utf8'))
             conn, addr = server.accept()
 
         except KeyboardInterrupt:
             print('\nClosing echo server...')
             break
 
+        # except StandardError:
+        #     conn.sendall(server_bad_connection().encode('utf8'))
+        #     break
     conn.close()
     server.close()
 
 
+def response_ok():
+    """Send a HTTP response to client."""
+    message = 'HTTP/1.1 200 OK\n'
+    if version_info[0] == 2:
+        message = message.decode('utf-8')
+    message += u"\r\n"
+    return message
+
+
+def response_error():
+    """Send a HTTP response to client if bad request."""
+    message = 'HTTP/1.1 500 Internal Server Error\n'
+    if version_info[0] == 2:
```

Comment:
No need to do a version check on this. A better approach to ensure all strings are unicode by using `from __future__ import unicode_literals`. This coerces all string literals to unicode in both versions. That means you can just explicitly and intentionaly encode to bytestring just once as you're sending the message through the connection.